autoload/neomake/makers/ft/markdown.vim

```diff
@@ -27,8 +27,15 @@ function! neomake#makers#ft#markdown#mdl() abort
 endfunction
 
 function! neomake#makers#ft#markdown#markdownlint() abort
+    let ver = neomake#compat#systemlist(['markdownlint', '--version'])
+    let ver_split = split(ver[0], '\.')
+    if len(ver_split) >= 2 && ver_split[0] == 0 && ver_split[1] <= 18
+        let errorformat = '%f: %l: %m' "old format
+    else
+        let errorformat = '%f:%l %m'
+    endif
     return {
-                \ 'errorformat': '%f: %l: %m'
+                \ 'errorformat': errorformat
                 \ }
 endfunction
```

Comment:
I think it is better to just handle both formats, making the extra version check unnecessary (which should get cached otherwise).

Reply:
Exactly! I updated it.